hermes/probe/create_file.go

```diff
@@ -0,0 +1,114 @@
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+type RandomFile struct {
+	Seed int64
+	Size int // File size in bytes
+}
+
+func (f *RandomFile) NewReader() *randomFileReader {
+	return &randomFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.Seed))}
+}
+
+type randomFileReader struct {
+	size int
+	// Current reading index.
+	i    int
+	rand *rand.Rand
+}
+
+func (r *randomFileReader) Read(b []byte) (n int, err error) {
+	if r.i >= r.size {
+		return 0, io.EOF
+	}
+	if len(b) > r.size-r.i {
+		n, err = r.rand.Read(b[:r.size-r.i])
+	} else {
+		n, err = r.rand.Read(b)
+	}
+	r.i += n
+	return
+}
+
+func (f *RandomFile) generateChecksum() (string, error) {
+	r := f.NewReader()
+	h := sha1.New()
+	buff := make([]byte, 8)
+	for {
```

Comment:
There's no need for this loop and the buffer.
io.Copy() takes care of everything.